mmif/serialize/mmif.py

```diff
@@ -511,7 +511,75 @@ def get_view_contains(self, at_types: Union[ThingTypesBase, str, List[Union[str,
                     return view
         return None
-    
+
+    def _is_in_time_between(self, start: Union[int, float], end: Union[int, float], annotation: Annotation) -> bool:
+        s, e = self.get_start(annotation), self.get_end(annotation)
+        return (s < start < e) or (s > start and e < end) or (s < end < e)
+
+    def _handle_time_unit(self, input_unit: str, ann_unit: str,
+                          start: int, end: int) -> Tuple[Union[int, float], Union[int,float]]:
+        if input_unit != ann_unit:
+            if input_unit == "seconds":
+                return start * 1000, end * 1000
+            else:
+                return start / 1000, end / 1000
+        return start, end
+
+    def get_annotations_between_time(self, start: int, end: int, time_unit: str = "milliseconds") -> Iterator[Annotation]:
+        """
+        Version: 1.0
+        Returns all 'Token' annotations aligned with 'TimeFrame' annotations sorted by start time within start and end time
+        Note: this function only works for mmif object obtained from Whisper-wrapper
+
+        :param start: the start time
+        :param end: the end time
+        :param time_unit: the time unit, either string "milliseconds" or "seconds", defaults to "milliseconds"
+        :return: a generator of 'Token' annotations
+        """
+        assert start <= end, "Start time must be less than end time"
+        assert start >= 0, "Start time must be greater than or equal to zero"
+        assert end >= 0, "End time must be greater than or equal to zero"
+        # 0. Initialize container and helper method
+        valid_tf_anns = []
+        tf_to_anns = defaultdict(list)
+
+        # 1. find all views that contain the type of TF
+        views = self.get_all_views_contain([AnnotationTypes.TimeFrame, AnnotationTypes.Alignment])
+
+        # 2. For each view, extract annotations that satisfy conditions that are TF/TP and fall into time interval
+        for view in views:
+            # Make sure time unit stay at the same level
+            start_time, end_time = self._handle_time_unit(time_unit, view.metadata.contains.get(AnnotationTypes.TimeFrame)["timeUnit"],
+                                                          start, end)
+            tf_anns = view.get_annotations(at_type=AnnotationTypes.TimeFrame)
+            al_anns = view.get_annotations(at_type=AnnotationTypes.Alignment)
+
+            # Select 'TimeFrame' annotations within given time interval
+            for tf in tf_anns:
+                if self._is_in_time_between(start_time, end_time, tf):
+                    valid_tf_anns.append(tf)
+
+            # Map 'TimeFrame' annotation to its aligned annotation
+            for align in al_anns:
+                source_id, target_id = align.get_property('source'), align.get_property('target')
+                try:
+                    source, target = view.get_annotation_by_id(source_id), view.get_annotation_by_id(target_id)
```

Comment:
This looks good. One thing I'd want to add here is usage of "long_id" of these annotation IDs. Something like this
``` python 
lengthen_id = lambda x: x if self.id_delimiter in x else f'{view.id}{self.id_delimiter}{x}'
source, target = map(lengthen_id, ...)
```
so that we can trace their parent views, just in case there's two or more views that hold the relevant annotations.